internal/v1/handler.go

```diff
@@ -486,6 +487,20 @@ func (h *Handlers) ComposeImage(ctx echo.Context) error {
 			logrus.Errorf("Unable to parse composer's compose response: %v", err)
 		} else {
 			_ = httpError.SetInternal(fmt.Errorf("%s", body))
+			var serviceStat composer.Error
+			if err := json.Unmarshal(body, &serviceStat); err != nil {
+				return httpError
+			}
+
+			id, err := strconv.Atoi(serviceStat.Id)
+			if err != nil {
+				return httpError
+			}
+
+			if id == 10 {
+				httpError.Message = "Error resolving OSTree repo"
+				httpError.Code = http.StatusBadRequest
+			}
```

Comment:
I don't think it's necessary to convert the id to an integer first. Just check if serviceStat.Id == "10"

Reply:
ok, done :)